File: lab-ui/src/api/container/modelService.ts
import request from "@/utils/request";
import type { QueryParams } from "@/utils/customType";
// request.defaults.baseURL = import.meta.env.VITE_APP_CONTAINER_API;
const containerPath = import.meta.env.VITE_APP_CONTAINER_API;
const adminPath = import.meta.env.VITE_APP_BASE_API + "/container";
const contextPath = adminPath + "/service";

interface InvokeDTO {
  /** 模型服务id */
  msId: string;
  /** 镜像id */
  imageId: string;
  /** 模型服务名称 */
  msName: string;
  /** mdl解析实体类 */
  modelClass: any;
  /** 是否部署完成 部署状态 INIT STARTED FINISHED ERROR */
  deployStatus: string;
  /** 服务类型 (0: 一次性的; 1: 后台运行)*/
  serviceType: string;
}
export type { InvokeDTO };

export function invokeService(modelService: InvokeDTO) {
  return request({
    url: contextPath + "/invoke",
    method: "post",
    data: modelService
  });
}

interface Service {
  imageId: string;
  containerId: string;
  msName: string;
  relativeDir: string;
  mdlFilePath: string;
  encapScriptPath: string;
}

export type { Service };

export function createModelService(service: Service) {
  return request({
    url: contextPath,
    method: "post",
    data: service,
    timeout: 2 * 60 * 1000
  });
}

export function selectServiceList(queryParams: QueryParams) {
  return request({
    url: contextPath + "/list",
    method: "get",
    params: queryParams
  });
}

export function getModelServiceById(msId: string) {
  return request({
    url: contextPath + "/" + msId,
    method: "get"
  });
}

File: lab-admin/src/main/java/org/opengms/admin/entity/dto/container/ModelService.java
package org.opengms.admin.entity.dto.container;

import lombok.Data;
import org.opengms.admin.entity.BaseEntity;
import org.opengms.admin.entity.dto.container.mdl.ModelClass;

/**
 * @author 7bin
 * @date 2022/11/12
 */
@Data
public class ModelService extends BaseEntity {

    /** 模型服务id */
    Long msId;

    /** 关联容器id ( 模型服务是以id为{containerId}的容器为基础环境发布的 ) */
    Long containerId;

    /** 镜像名 */
    // String imageName;

    /** 镜像id 模型基于该镜像启动运行环境 */
    Long imageId;

    /** 模型服务名称 */
    String msName;

    /** 原container中model存放路径  */
    String relativeDir;

    /** mdl文件路径 */
    String mdlFilePath;

    /** 封装脚本路径 */
    String encapScriptPath;

    /** mdl解析实体类 */
    ModelClass modelClass;

    /** 是否部署完成 部署状态 INIT STARTED FINISHED ERROR */
    String deployStatus;

    /** 是否基于当前环境创建新环境 */
    // Boolean newEnv;

    /** 新部署包的id */
    // String pkgId;

    /** 部署包路径 */
    // String deployPkgPath;

    /** 导出的镜像包名 */
    // String imageTar;

    /** 服务类型 (0: 运行完立即销毁; 1: 后台运行) */
    String serviceType;

}
